server/game/gameSystems/DiscardHandSystem.ts

```diff
@@ -0,0 +1,32 @@
+import { EventName, Location } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import * as Helpers from '../core/utils/Helpers.js';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface IDiscardHandSystemProperties extends IPlayerTargetSystemProperties {}
+
+/**
+ * A {@link GameSystem} which discards a player's entire hand
+ */
+export class DiscardHandSystem<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardHandSystemProperties> {
```

Comment:
nit: just to be brutally clear let's maybe call it like `DiscardWholeHand` or `DiscardEntireHand`